src/test/java/com/spotify/docker/client/auth/MultiRegistryAuthSupplierTest.java

```diff
@@ -0,0 +1,127 @@
+/*-
+ * -\-\-
+ * docker-client
+ * --
+ * Copyright (C) 2016 - 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.docker.client.auth;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.spotify.docker.client.messages.RegistryAuth;
+import com.spotify.docker.client.messages.RegistryConfigs;
+import org.junit.Test;
+
+public class MultiRegistryAuthSupplierTest {
+
+  private final RegistryAuthSupplier supplier1 = mock(RegistryAuthSupplier.class);
+  private final RegistryAuthSupplier supplier2 = mock(RegistryAuthSupplier.class);
+
+  private final RegistryAuthSupplier multiSupplier =
+      new MultiRegistryAuthSupplier(ImmutableList.of(supplier1, supplier2));
+
+  @Test
+  public void testAuthFor() throws Exception {
+    final String image1 = "foobar:latest";
+    final RegistryAuth auth1 = RegistryAuth.builder().build();
+    when(supplier1.authFor(image1)).thenReturn(auth1);
+
+    assertThat(multiSupplier.authFor(image1), is(auth1));
+    verify(supplier2, never()).authFor(image1);
```

Comment:
Is this important to test? ie that the sub-suppliers are in a list that's traversed in order? Test needs to know the implementation of MultiRegistryAuthSupplier rather than just checking the correct auth is returned.